### PR TITLE
[Fix] make thrust temp memory allocation managed by HKV allocator.

### PR DESCRIPTION
**Pull request by rhdong**

### CODE REVIEW

**Review by LinGeLin — APPROVED** [review]
LGTM

File: include/merlin/core_kernels/upsert.cuh [hunk 1]
```diff
@@ -1146,6 +1146,7 @@ struct KernelSelector_Upsert {
       }
     };
 
+#if defined(CUDART_VERSION) && (CUDART_VERSION >= 11030)
```
Comment (rhdong) [inline include/merlin/core_kernels/upsert.cuh:1149]: Hi @jiashuy, I cleaned some TLPv2 warnings we discussed before. Please help review and confirm. Thanks!

**Pull request merged**